backend/src/api/googleAuth.ts

```diff
@@ -0,0 +1,126 @@
+import express, { Request, Response } from 'express'
+import { DI } from '@/di'
+import { GoogleOAuthService } from '@/lib/googleOAuth'
+import { generateAccessToken } from '@/lib/tokenUtils'
+import { ENABLE_SECURITY_SETTINGS, FRONTEND_URL } from '@/settings'
+import { posthogCapture } from '@/lib/posthogUtils'
+import crypto from 'crypto'
+
+export const googleAuthRouter = express.Router({ mergeParams: true })
+
+const initiateGoogleOAuth = async (req: Request, res: Response) => {
+    try {
+        const googleOAuth = new GoogleOAuthService()
+
+        const state = crypto.randomBytes(32).toString('hex')
+
+        res.cookie('oauth_state', state, {
+            httpOnly: true,
+            secure: ENABLE_SECURITY_SETTINGS,
+            sameSite: 'lax',
+            maxAge: 10 * 60 * 1000,
+            path: '/',
+        })
+
+        const authUrl = googleOAuth.getAuthUrl(state)
+        res.redirect(authUrl)
+    } catch (error) {
+        console.error('Error initiating Google OAuth:', error)
+        res.redirect(`${FRONTEND_URL}/login?error=oauth_init_failed`)
+    }
+}
+
+const handleGoogleCallback = async (req: Request, res: Response) => {
+    try {
+        const { code, state, error } = req.query
+
+        if (error) {
+            return res.redirect(`${FRONTEND_URL}/login?error=access_denied`)
+        }
+
+        if (!code || typeof code !== 'string') {
+            return res.redirect(`${FRONTEND_URL}/login?error=missing_code`)
+        }
+
+        const storedState = req.cookies.oauth_state
+        if (!storedState || storedState !== state) {
+            return res.redirect(`${FRONTEND_URL}/login?error=invalid_state`)
+        }
+
+        res.clearCookie('oauth_state')
+
+        const googleOAuth = new GoogleOAuthService()
+        const googleUser = await googleOAuth.verifyIdToken(code)
+
+        if (!googleUser.email_verified) {
+            return res.redirect(`${FRONTEND_URL}/login?error=email_not_verified`)
+        }
+
+        const normalizedEmail = googleUser.email.toLowerCase().trim()
+
+        let user = await DI.users.findOne({ email: normalizedEmail })
+
+        if (!user) {
+            user = DI.users.create({
+                email: normalizedEmail,
+                password: '',
+                emailVerified: true,
+                first_name: googleUser.given_name,
+                last_name: googleUser.family_name,
+                googleId: googleUser.sub,
+                authProvider: 'google',
+                profilePicture: googleUser.picture,
+                is_superuser: false,
+                is_staff: false,
+                is_active: true,
+                date_joined: new Date(),
+            })
+
+            await DI.em.persistAndFlush(user)
+
+            posthogCapture('user_signed_up_google', user.email, {
+                user_email: user.email,
+                oauth_provider: 'google',
+            })
```

Comment:
this should ideally be the same event name as our normal signup, and then as you've done here oauth_provider specifies that this is google. 

or we could add a auth_method prop that's 'password' | 'google' | 'github" etc